### ISSUE TITLE
Difference between this and node-inspector

### ISSUE THREAD

**Issue by amcdnl** [buggerjs/bugger#67]
It would be nice to know what the difference between this and Node Inspector is. I can tell some UI differences at first glance.

**Comment by jkrems** [comment]
**TL;DR:** It's roughly the same but `node-inspector` is more stable & has an actual company backing it. From a user perspective, it's roughly the same.

#### Features

##### `bugger`

* `chrome://inspect` support
* More detailed network timings
* True repl in console, including support for `require`
* Support for `child_process#fork` / workers

##### `node-inspector

* Preloading files
* Remote debugging without having to be installed on the remote host
* Support for saving files when they were live-edited (`bugger` assumes you'd just use the workspace feature, so this isn't *really* a feature `bugger` lacks)

#### Random Notes

* `node-inspector` is widely used and has some sort of test coverage; neither is really true for `bugger`
* `node-inspector` uses a forked version of the DevTools UI, `bugger` just implements the protocol and uses the "official" UI that is part of Chrome
* `node-inspector` runs in a separate process (possibly even on a separate machine) and uses the V8 debug protocol; `bugger` runs as part of the app and uses an IO thread to dispatch messages
  - `node-inspector` supports remote debugging of a "vanilla" node process
  - `bugger` is not constraint by the V8 debug protocol (which Chrome itself stopped using)
* `node-inspector` cares about good support for older node versions, `bugger` supports node 4 and pretty much nothing else at this point.

Hope this clears things up. :)

**Comment by amcdnl** [comment]
Thanks for the feedback! So what was the motivation for you to create this over use node-inspector?

**Comment by jkrems** [comment]
Short answer: Understanding how Chrome's debugger and V8 internals work. Also, I wanted a clean implementation of the devtools protocol for node so that I could use it as a backend for debug bindings in editors that would work for both remotely debugging in Chrome and node. `node-inspector` bundling UI and node bindings into one is great if you want to use the devtools UI. Not so great if you want to develop an atom.io or VSCode plugin.

**Comment by amcdnl** [comment]
Ah, I wondered why no one had embedded node inspector into a IDE plugin. Look forward to seeing yours ;)
